### PR TITLE
ENG-14452-Complete the new sqlcmd command with auto-completion

### PR DESCRIPTION
**Pull request by Chaozz**
- I reconsidered packaging and releasing, I think it is better to leave the install command to let the user install it themselves. It is quite simple and python-standard, and users can decide where they want to install(virtualenv or not).

- Jenkins test is enabled. Here are examples of a [good result](http://ci.voltdb.lan/view/Branch-jobs/view/branch-ENG-14452-voltsql-sqlcmd-with-auto-completion/job/branch-2-sqlcmdtest-ENG-14452-voltsql-sqlcmd-with-auto-completion/lastSuccessfulBuild/) and a [failure](http://ci.voltdb.lan/view/Branch-jobs/view/branch-ENG-14452-voltsql-sqlcmd-with-auto-completion/job/branch-2-sqlcmdtest-ENG-14452-voltsql-sqlcmd-with-auto-completion/lastFailedBuild/). You can see the details in console output.

copy&paste from readme:

voltcli
=======

Install Requirements
----------------

- Install Python 2

- Open a terminal, go to the project folder, and run this command to install all project dependencies:

    ```
    $ pip install -r requirements.txt
    ```
    
- Or if you want to install the dependencies offline, you can use

    ```
    $ pip install -r requirements_offline.txt
    ```
      
Run
---
In project root folder, simply run:
```
$ python voltsql/voltsql.py
```

Or
```
$ ./voltsql.sh
```

Test
----
```
$ tox
```

Architecture
------------
```text
+-----------------------------------+
|    _    __      ____  ____  ____  |	
|   | |  / /___  / / /_/ __ \/ __ ) |
|   | | / / __ \/ / __/ / / / __  | |
|   | |/ / /_/ / / /_/ /_/ / /_/ /  |
|   |___/\____/_/\__/_____/_____/   |
+-----------------------------------+
           |         ^
           v         | 
+---------------------------------------------+ 
|        SQLCMD                               |
|     ======	                              |
|   We still use the SQLCMD to execute SQLs,  |
|   and get response from its stdout.         |
|   It's like our "server"                    |
+---------------------------------------------+
           |         ^
           v         |
+---------------------------------------------------------------+
|     VoltSQL                                                   |
|     =======                                                   |
|          - Build interactive command line based on Python     |
|            Prompt Toolkit.                                    |
|          - Get the database information from @SystemCatalog   |
|          - Parse the buffer while typing, so that we can      |
|            generate context-sensitive suggestions based on    |
|            current cursor's postion                           |
|                                                               |
|            +-----------------------------------------------+  |
|            | Completer, SQLParser, refresher, Executor     |  |
|            | =========================================     |  |
|            |                                               |  |
|            +-----------------------------------------------+  |
+---------------------------------------------------------------+
```

### CODE REVIEW

**Comment by Chaozz** [comment]
I need to figure out how to configure my test in Jenkins, I think it is similar to the vdm test.

File: lib/python/voltsql/voltsql/voltexecutor.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        self.parameters = {"host": server, "port": port, "procedure_timeout": query_timeout}
+        if user:
+            self.parameters["username"] = user
+        if password:
+            self.parameters["password"] = password
+
+        if credentials:
+            # if credential file is specified, parse it as username/password
+            try:
+                with open(credentials, "r") as credentialsFile:
```
Comment (Chaozz) [inline lib/python/voltsql/voltsql/voltexecutor.py]: Our python client doesn't support credentials, I believe we should, and it just a trivial change.
I may create a ticket for that.

**Comment by yzhang1991** [comment]
@ajgent Can you help review the README? Thanks!

File: bin/voltsql [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+# that are loaded from <name>.d subdirectories. It loads the version number
+# from version.txt in the script's parent directory. It can be copied to other
+# names, and also to other locations if the path-building is adjusted. The
+# description should also be changed if re-used for another base command.
+import sys
+import os
+
+try:
+    # ensure version 2.6+ of python
+    if sys.version_info[0] == 2 and sys.version_info[1] < 6:
```
[the remark this answers is not in the document]
Comment (Chaozz) [reply bin/voltsql]: We don't support Python3, I have updated the readme.

File: bin/voltsql [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        for dir in os.environ['PATH'].split(':'):
+            for name in ('python2.7', 'python2.6'):
+                path = os.path.join(dir, name)
+                if os.path.exists(path):
+                    print 'Re-running with %s...' % path
+                    os.execv(path, [path] + sys.argv)
```
[the remark this answers is not in the document]
Comment (Chaozz) [reply bin/voltsql]: done

File: bin/voltsql [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    version     = open(os.path.join(base_dir, 'version.txt')).read().strip()
+    description = 'This is the command line interface to VoltDB administrative functions.'
+    standalone  = True
+    # Tweak the Python library path to call voltcli.runner.main().
+    # Possible installed library locations.
+    if os.path.isdir('/opt/lib/voltdb/python/voltsql'):
+        sys.path.insert(0, '/opt/lib/voltdb/python/voltsql')
+    if os.path.isdir('/usr/share/lib/voltdb/python/voltsql'):
+        sys.path.insert(0, '/usr/share/lib/voltdb/python/voltsql')
+    if os.path.isdir('/usr/lib/voltdb/python/voltsql'):
```
[the remark this answers is not in the document]
Comment (Chaozz) [reply bin/voltsql]: done

File: lib/python/voltsql/tests/test_naive_completion.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+def test_function_name_completion(completer, complete_event):
+    text = 'SELECT MA'
+    position = len('SELECT MA')
+    result = set(completer.get_completions(
+        Document(text=text, cursor_position=position),
+        complete_event))
+    print(result)
+    assert result == {Completion(text=u'MAKEVALIDPOLYGON', start_position=-2),
+                      Completion(text=u'MAXEXTENTS', start_position=-2),
```
[the remark this answers is not in the document]
Comment (Chaozz) [reply lib/python/voltsql/tests/test_naive_completion.py]: I removed them from `literals.json`

**Pull request merged**